src/features/profile/Profile.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useSelector } from "react-redux";
+import useProfileState from "./use-profile-state";
+import { AuthState } from "features/auth/store/types";
+import { fetchOneProfile } from "./services/profile.service";
+import ShowSkills from "./components/show-skills";
+import ShowExperiences from "./components/show-experiences";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+
+const Profile = () => {
+  const { profile, setProfile } = useProfileState();
+
+  const id = parseInt(useParams<RouteParams>().id!);
+
+  function useAuthState(): AuthState {
+    return useSelector<{ auth: AuthState }>((state) => state.auth) as AuthState;
+  }
+  const { token } = useAuthState();
+  const loggedUser = useAuthState().user;
+
+  const getProfile = useCallback(async () => {
+    try {
+      if (token && id) {
+        const data = await fetchOneProfile(token, id);
+        setProfile(data);
+        console.log(token);
+      }
```

Comment:
Evitemos niveles de abstraccion innecesarios, investiga sobre `Guard Clausures`, imagen que lo resume:

![image](https://github.com/luizzzito/ER-frontend/assets/22270579/8be4817c-f632-46bd-a463-2b94da14d69d)

```suggestion
      if (!token || !id) return;

      const data = await fetchOneProfile(token, id);
      setProfile(data);
      console.log(token);
```

Reply:
Si preguntas por que cambie de `if (A && B)` hacia `if (!A || !B)` es por la **Ley de De Morgan** 
```
¬(A∧B)=(¬A)∨(¬B)
```
Hijo.

Reply:
oka